### PR TITLE
Fix error compositing images into vector tiles

### PR DESCRIPTION
**Pull request by JesseCrocker**
Was getting an error trying to merge a vector tile and a png tile.

I think this might have broken with node-mapnik 3.5.

See comment on commit for error message from error this is fixing.

### CODE REVIEW

**Comment by sethfitz** [comment]
Yeah, that makes sense. Does this bump the Mapnik version requirement to `^3.5.0`?

File: index.js [hunk 1]
```diff
@@ -135,14 +135,13 @@ module.exports = function(tilelive, options) {
...
-              function(img, done) {
-                vt.addImage(img, src.info.id.replace(/\./g, "_"));
-
-                return done(null, vt, headers);
+            im.encode("webp", function(err, encoded) {
+              if (err) {
+                return next(err);
               }
-            ], next);
+              vt.addImageBufferSync(encoded, src.info.id.replace(/\./g, "_"));
```
Comment (sethfitz) [inline index.js:142]: Is there an async variant of this? If so, we should probably use it.

File: index.js [hunk 2]
```diff
@@ -135,14 +135,13 @@ module.exports = function(tilelive, options) {
               return next(err);
             }
 
-            return async.waterfall([
-              async.apply(im.encode, "webp"),
-              function(img, done) {
-                vt.addImage(img, src.info.id.replace(/\./g, "_"));
```
Comment (sethfitz) [inline index.js:141]: It looks like the cleaner change is just to change `vt.addImage` to `vt.addImageBufferSync`, as the `webp` encode + error handling was taken care of by `async.waterfall`.

Comment (sethfitz) [inline index.js:141]: Oh, n/m. This is what the error is related to. The fix might be to bind the image to `encode`:

``` javascript
async.apply(im.encode.bind(im), "webp"),
```

**Comment by sethfitz** [comment]
@JesseCrocker bump?! Are you running this change in production?

**Comment by JesseCrocker** [comment]
No, not using it in production, and im not sure this works right, so i'll close this.

**Pull request closed without merging**